### PR TITLE
Add openAPIV3Schema to CRDs

### PR DESCRIPTION
**Pull request by sanchezl**
Add openAPIV3Schema to CRD for custom resource validation.

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -43,6 +44,16 @@ GOPATH=${WORK_DIR}/go/ go install k8s.io/code-generator/cmd/client-gen
 GOPATH=${WORK_DIR}/go/ go install k8s.io/code-generator/cmd/deepcopy-gen
 popd
 
+git clone --branch v0.2.4 --depth 1 -c advice.detachedHead=false \
```
Comment (sttts) [inline hack/update-codegen.sh]: will this fail on second invocation?

Comment (sanchezl) [reply hack/update-codegen.sh]: No. Each invocation uses a new `${WORK_DIR}`.

File: manifests/storage_state_crd.yaml [hunk 2]
```diff
@@ -17,3 +17,63 @@ spec:
   - name: v1alpha1
     served: true
     storage: true
+  "validation":
+    "openAPIV3Schema":
```
Comment (caesarxuchao) [inline manifests/storage_state_crd.yaml]: Do you know why these two lines are quoted?

**Comment by caesarxuchao** [comment]
For documentation purpose, can you say in the first comment on why we want to add the openAPIV3Schema?

/approve
feel free to label it when you update the comment.

File: manifests/storage_state_crd.yaml (hunk 2 above)
Comment (sttts) [reply manifests/storage_state_crd.yaml]: Some encoding issue of the yaml library. We saw that before with controller-gen. Maybe @directxman12 has an idea.

**Pull request merged**